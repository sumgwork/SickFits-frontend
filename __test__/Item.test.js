import { shallow } from "enzyme";
import Item from "../components/Item";
import toJSON from "enzyme-to-json";

const fakeItem = {
  id: "ABC",
  title: "A cool item",
  description: "This item is cool!",
  image: "dog.jpg",
  largeImage: "largeDog.jpg",
  price: 5000
};

describe("<Item />", () => {
  it("should render the image properly", () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const img = wrapper.find("img");
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
    // console.log(wrapper.debug());
  });
  it("should render title and price properly", () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const PriceTag = wrapper.find("PriceTag");
    expect(PriceTag.children().text()).toBe("A$50");
    expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
  });

  it("should render out the buttons properly", () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const buttonList = wrapper.find(".buttonList");
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find("Link").exists()).toBeTruthy();
    expect(buttonList.find("AddToCart").exists()).toBeTruthy();
    expect(buttonList.find("DeleteItem").exists()).toBe(true);
  });

  it("renders and matches the snapshot", () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
