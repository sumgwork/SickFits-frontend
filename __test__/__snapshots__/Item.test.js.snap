// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    A$50
  </PriceTag>
  <p>
    This item is cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id={
        Object {
          "description": "This item is cool!",
          "id": "ABC",
          "image": "dog.jpg",
          "largeImage": "largeDog.jpg",
          "price": 5000,
          "title": "A cool item",
        }
      }
    />
    <DeleteItem
      id="ABC"
    >
      Delete item ❎
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
